### PR TITLE
index fields by name, original name, and full name

### PR DESCRIPTION
**Pull request by acharis**
like mysql-connector-j does

@bbeaudreault @harshit-gangal

### CODE REVIEW

File: java/client/src/main/java/io/vitess/client/cursor/FieldMap.java [hunk 1]
```diff
@@ -37,20 +37,36 @@
...
-      String columnLabel = field.getName();
       // If multiple columns have the same name,
       // prefer the earlier one as JDBC ResultSet does.
-      if (!indexMap.containsKey(columnLabel)) {
-        indexMap.put(columnLabel, columnIndex);
+      String columnLabel = field.getName();
+      if (!labelMap.containsKey(columnLabel)) {
+        labelMap.put(columnLabel, columnIndex);
+      }
+      String origName = field.getOrgName();
```
Comment (bbeaudreault) [inline java/client/src/main/java/io/vitess/client/cursor/FieldMap.java]: the Field may not have these extra values. This is the case when IncludedFields != ALL. So I would check whether the value is null/empty before doing the extra map work and such. Same goes for both new maps.

Comment (bbeaudreault) [inline java/client/src/main/java/io/vitess/client/cursor/FieldMap.java]: the Field may not have these extra values. This is the case when IncludedFields != ALL. So I would check whether the value is null/empty before doing the extra map work and such. Same goes for both new maps.

**Comment by acharis** [comment]
@sougou @bbeaudreault i got around to adding a test to this, i think we're ready.
(what's netlify?)

**Comment by derekperkins** [comment]
Netlify is for automated deployment of vitess.io. We're still working on it, so you can ignore it for now.

**Comment by bbeaudreault** [comment]
LGTM

**Pull request merged**